### DISCUSSION TITLE
can't get redirect() working after a delete request

### DISCUSSION THREAD

**Discussion by mehdirayan** [adonisjs/core#4775] [Help] [answered]
I have in my front-end a list of contacts. I need to delete a contact and refresh the page.
i'm using inertia
so in my contact.vue I write this:

const supprimerContact = (data:Contact)=>{

    router.delete("contacts/", {data:{contactData:data,clientData:props.clientData},

})

in my controller i write this

 public async destruct({ request, inertia , response }: HttpContext) {

    const contact: Contact = request.all().contactData
    const client:Client = request.all().clientData

    await ContactsService.deleteContact(contact.id)

    let newContact = await ContactsService.getContacts(client.id)

    //return inertia.render('clients/contacts', { clientData:client,contactData: newContact })
   
   response.redirect().toRoute('clients/clients', { clientData:client,contactData: newContact })
    
  }
}

and this is my routes

router.group(()=>{

  router.get('contacts', [ContactsController,'show'])

  router.delete('contacts',[ContactsController,'destruct'])

}).prefix('/contacts')

I got this error "Cannot lookup route "clients/clients"" and i dont understand why adonis can't find the route

**Comment by RomainLanz** [comment]
Hey @mehdirayan! 👋🏻 

The router is looking for a route with the name `clients/clients`. If you want to redirect to the URL `clients/clients` then you have to use `toPath`.

📚 https://docs.adonisjs.com/guides/basics/response#redirects

**Reply by mehdirayan** [reply]
hi
sorry i wasn't using well the redirect function. so I rewrite it like this:
 public async destruct({ request, response }: HttpContext) {

    const contact: Contact = request.all().contactData
    const client: Client = request.all().clientData


    await ContactsService.deleteContact(contact.id)


    response.redirect().toRoute('contacts.contacts.show', { clientData: client })

  }
}

now it redirect to contacts/contacts with a get request(status;303) **but without query parameters**

what i'm doing wrong ?

**Reply by mehdirayan** [reply, marked as answer]
hi every body
this is working for me
public async destruct({ request, response }: HttpContext) {

    const contact: Contact = request.all().contactData
    const client: Client = request.all().clientData


    await ContactsService.deleteContact(contact.id)

    
    return response.redirect().withQs({ clientData: client}).toRoute('contacts.contacts.show')

  }

The redirect alone doesn't send query parameters. i had to add withQs() function.
thanks for help!
